benchcab/config.py

```diff
@@ -1,170 +1,93 @@
 """A module containing all *_config() functions."""
-
 from pathlib import Path
 
 import yaml
-
 from benchcab import internal
+from cerberus import Validator
+import benchcab.utils as bu
+
+
+class ConfigValidationException(Exception):
+
+    def __init__(self, validator: Validator):
+        """Config validation exception.
+
+        Parameters
+        ----------
+        validator: cerberus.Validator
+            A validation object that has been used and has the errors attribute.
+        """
+
+        # Nicely format the errors.
+        errors = [f'{k} = {v}' for k, v in validator.errors.items()]
 
+        # Assemble the error message and 
+        msg = '\n\nThe following errors were raised when validating the config file.\n'
+        msg += '\n'.join(errors) + '\n'
 
-def check_config(config: dict):
-    """Performs input validation on config file.
+        # Raise to super.
+        super().__init__(msg)
 
-    If the config is invalid, an exception is raised. Otherwise, do nothing.
+
+def validate_config(config: dict) -> bool:
+    """Validate the configuration dictionary.
+
+    Parameters
+    ----------
+    config : dict
+        Dictionary of configuration loaded from the yaml file.
+
+    Returns
+    -------
+    bool
+        True if valid, exception raised otherwise.
+
+    Raises
+    ------
+    ConfigValidationException
+        Raised when the configuration file fails validation.
+    """
+    
+    # Load the schema
+    schema = bu.load_package_data('config-schema.yml')
+
+    # Create a validator
+    v = Validator(schema)
+
+    # Validate
+    is_valid = v.validate(config)
+    
+    # Valid
+    if is_valid:
+        return True
+
+    # Invalid
+    raise ConfigValidationException(v)
+
+
+def read_config(config_path: str) -> dict:
+    """Reads the config file and returns a dictionary containing the configurations.
+
+    Parameters
+    ----------
+    config_path : str
+        Path to the configuration file.
+
+    Returns
+    -------
+    dict
+        Configuration dict.
+    
+    Raises
+    ------
+    ConfigValidationError
+        Raised when the configuration file fails validation.
     """
-    if any(key not in config for key in internal.CONFIG_REQUIRED_KEYS):
```

Comment:
We can probably remove the `CONFIG_REQUIRED_KEYS` constant from internal.py

https://github.com/CABLE-LSM/benchcab/blob/0b36493f81fbd75c89829c8dc4d65ac6d7968609/benchcab/internal.py#L9